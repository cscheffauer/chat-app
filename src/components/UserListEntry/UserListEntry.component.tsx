import React from 'react';

import './UserListEntry.scss';

interface Props {
	key: String;
	username: String;
}

const UserListEntry: React.FunctionComponent<Props> = ({ username }) => {
	return (
		<div className='userlistentry'>
			<p className='username'>{username}</p>
		</div>
	);
};

export default UserListEntry;
