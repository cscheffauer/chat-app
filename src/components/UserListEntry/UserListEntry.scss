@import '../../variables';

.userlistentry {
	width: 100%;
	height: 8vh;
	display: table;
	border-bottom: $border-style-tabs;
	p {
		display: table-cell;
		vertical-align: middle;
		padding-left: 5vw;
	}
}
