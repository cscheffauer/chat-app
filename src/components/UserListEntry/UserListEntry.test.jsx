import React from 'react';
import { mount } from 'enzyme';
import UserListEntry from './UserListEntry.component';

const username = 'Max Mustermann';

describe('<UserListEntry />', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = mount(<UserListEntry key={'1'} username={username} />);
	});

	afterEach(() => {
		jest.clearAllMocks();
	});
	it('expect to render UserListEntry component', () => {
		expect(wrapper).toMatchSnapshot();
	});

	describe('Testing the UserListEntry', () => {
		it('expect to have the username set as a text in the entry', () => {
			expect(wrapper.text()).toBe(username);
		});
	});
});
