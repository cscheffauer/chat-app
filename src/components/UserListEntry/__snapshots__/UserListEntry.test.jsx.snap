// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserListEntry /> expect to render UserListEntry component 1`] = `
<UserListEntry
  username="Max Mustermann"
>
  <div
    className="userlistentry"
  >
    <p
      className="username"
    >
      Max Mustermann
    </p>
  </div>
</UserListEntry>
`;
